contracts/utils/PausableUntil.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+import {IERC6372} from "@openzeppelin/contracts/interfaces/IERC6372.sol";
+
+/**
+ * @title Pausable
+ * @author @CarlosAlegreUr
+ *
+ * @dev Contract module which allows children to implement an emergency stop
+ * mechanism that can be triggered by an authorized account.
+ *
+ * Stops can be of undefined duration or for a certain amount of time.
+ *
+ * This module is used through inheritance. It will make available the
+ * modifiers `whenNotPaused` and `whenPaused`, which can be applied to
+ * the functions of your contract. Note that they will not be `Pausable` by
+ * simply including this module, only once the modifiers are put in place
+ * and access to call the internal `_pause()`, `_unpause()`, `_pauseUntil()`
+ * functions is coded.
+ *
+ * [ ⚠️ WARNING ⚠️ ]
+ * This version should be backwards compatible with previous OpenZeppelin `Pausable`
+ * versions as it uses the same 1 storage slot in a backwards compatible way.
+ *
+ * However this has not been tested yet. Please test locally before updating any
+ * contract to use this version.
+ */
+abstract contract PausableUntil is Pausable, IERC6372 {
+    /**
+     * @dev Storage slot is structured like so:
+     *
+     * - Least significant 8 bits: signal pause state.
+     *   1 for paused, 0 for unpaused.
+     *
+     * - After, the following 48 bits: signal timestamp at which the contract
+     *   will be automatically unpaused if the pause had a duration set.
+     */
+    uint48 private _pausedUntil;
+
+    /**
+     * @dev Emitted when the pause is triggered by `account`. `unpauseDeadline` is 0 if the pause is indefinite.
+     */
```

Comment:
This is now incorrect. If `unpauseDeadline` is 0 (or less than `block.timestamp`) the contract is not paused at all.